include/ensmallen_bits/age_moea/agemoea_impl.hpp

```diff
@@ -0,0 +1,781 @@
+AGEMOEA::EvaluateObjectives(
+    std::vector<MatType>&,
+    std::tuple<ArbitraryFunctionType...>&,
+    std::vector<arma::Col<typename MatType::elem_type> >&)
+{
+  // Nothing to do here.
+}
+
+//! Evaluate the objectives for the entire population.
+template<std::size_t I,
+         typename MatType,
+         typename ...ArbitraryFunctionType>
+typename std::enable_if<I < sizeof...(ArbitraryFunctionType), void>::type
+AGEMOEA::EvaluateObjectives(
+    std::vector<MatType>& population,
+    std::tuple<ArbitraryFunctionType...>& objectives,
+    std::vector<arma::Col<typename MatType::elem_type> >& calculatedObjectives)
+{
+  for (size_t i = 0; i < populationSize; i++)
+  {
+    calculatedObjectives[i](I) = std::get<I>(objectives).Evaluate(population[i]);
+    EvaluateObjectives<I+1, MatType, ArbitraryFunctionType...>(population, objectives,
+                                                               calculatedObjectives);
+  }
+}
+
+//! Reproduce and generate new candidates.
+template<typename MatType>
+inline void AGEMOEA::BinaryTournamentSelection(std::vector<MatType>& population,
+                                             const MatType& lowerBound,
+                                             const MatType& upperBound)
+{
+  std::vector<MatType> children;
+
+  while (children.size() < population.size())
+  {
+    // Choose two random parents for reproduction from the elite population.
+    size_t indexA = arma::randi<size_t>(arma::distr_param(0, populationSize - 1));
+    size_t indexB = arma::randi<size_t>(arma::distr_param(0, populationSize - 1));
+
+    // Make sure that the parents differ.
+    if (indexA == indexB)
+    {
+      if (indexB < populationSize - 1)
+        indexB++;
+      else
+        indexB--;
+    }
+
+    // Initialize the children to the respective parents.
+    MatType childA = population[indexA], childB = population[indexB];
+
+    if(arma::randu() <= crossoverProb)
+      Crossover(childA, childB, population[indexA], population[indexB], 
+                lowerBound, upperBound);
+
+    Mutate(childA, 1.0 / static_cast<double>(numVariables),
+          lowerBound, upperBound);
+    Mutate(childB, 1.0 / static_cast<double>(numVariables),
+          lowerBound, upperBound);
+
+    // Add the children to the candidate population.
+    children.push_back(childA);
+    children.push_back(childB);
+  }
+
+  // Add the candidates to the elite population.
+  population.insert(std::end(population), std::begin(children), std::end(children));
+}
+
+//! Perform simulated binary crossover (SBX) of genes for the children.
+template<typename MatType>
+inline void AGEMOEA::Crossover(MatType& childA,
+                             MatType& childB,
+                             const MatType& parentA,
+                             const MatType& parentB,
+                             const MatType& lowerBound,
+                             const MatType& upperBound)
+{
+    //! Generates a child from two parent individuals
```

Comment:
It seems like the indentation in alignment both in this function definition and the function body itself are off.